frontend/src/components/CourseSearchList.tsx

```diff
@@ -23,12 +23,16 @@ const CoursePage = () => {
 
   const coursesToShow: string[] = useMemo(() => {
     if (page === 1 && exactResultsCourses.length > 0) {
-      return [
-        ...exactResultsCourses,
-        ...pageCourses.filter(
-          (courseID) => !exactResultsCourses.includes(courseID)
-        ),
-      ];
+      const filteredCourses = pageCourses.filter(
+        (courseID) => !exactResultsCourses.includes(courseID)
+      );
+
+      if(pageCourses.includes(exactResultsCourses[0])) {
```

Comment:
Since `filteredCourses` is only required in the `if` case, let's try and optimize this by moving the `filteredCourses` inside the `if` case. It might be expensive to compute `filteredCourses`.